frontend/public/components/utils/webhooks.tsx

```diff
@@ -56,7 +56,26 @@ export const WebhookTriggers: React.FC<WebhookTriggersProps> = (props) => {
   });
   const tableColumnClasses = getTableColumnClasses(canGetSecret);
   const webhooks = _.filter(triggers, (trigger) => webhookTriggers.has(trigger.type));
-  if (_.isEmpty(webhooks)) {
+  const [webhookSecrets, setWebhookSecrets] = React.useState([])
+
+  const webhookNames = _.map(webhooks, (webhook) => {
+    const triggerProperty = getTriggerProperty(webhook);
+    return _.get(webhook, [triggerProperty, 'secretReference', 'name']);
+  })
+
+  React.useEffect(() => {
+    Promise.all(
+      _.map(webhookNames, (webhookName) => {
+        return k8sGet(SecretModel, webhookName, namespace).then(res => {
+          return res
+        });
+      })
+    ).then(response => {
+      setWebhookSecrets(response)
+    })
+  });
```

Comment:
You need to specify dependencies or you'll request the secrets again on every render